src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -0,0 +1,158 @@
+            }
+            throw ThrowHelper.GetInvalidOperationException(ThrowHelper.ExceptionResource.PinNotOpenedError, pin:pinNumber);
+        }
+
+        protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            // Libgpiod Api do not support pull up or pull down resistors for now.
+            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            SafeLineHandle pinHandle = Interop.GetChipLineByOffset(_chip, pinNumber);
+            if (pinHandle == null)
+            {
+                ThrowHelper.ThrowIOException(ThrowHelper.ExceptionResource.OpenPinError, Marshal.GetLastWin32Error());
+            }
+
+            _pinNumberToSafeLineHandle.Add(pinNumber, pinHandle);
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                int result = Interop.GetGpiodLineValue(pinHandle);
+                if (result == -1)
+                {
+                    ThrowHelper.ThrowIOException(ThrowHelper.ExceptionResource.ReadPinError, Marshal.GetLastWin32Error(), pinNumber);
+                }
+                return result;
+            }
+            throw ThrowHelper.GetInvalidOperationException(ThrowHelper.ExceptionResource.PinNotOpenedError, pin: pinNumber);
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            int requestResult  = -1;
+            string consumer = pinNumber.ToString();
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
+            {
+                if (mode == PinMode.Input)
+                {
+                    requestResult = Interop.RequestLineInput(pinHandle, consumer);
+                }
+                else
+                {
+                    requestResult = Interop.RequestLineOutput(pinHandle, consumer);
+                }
+            }
+
+            if (requestResult == -1)
+            {
+                ThrowHelper.ThrowIOException(ThrowHelper.ExceptionResource.SetPinModeError, Marshal.GetLastWin32Error(), pinNumber);
+            } 
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+            if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
```

Comment:
nit: invert this (which would avoid chances of bugs with missing return statements)

```C#
            if (!_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
            {
               throw ThrowHelper.GetInvalidOperationException(ThrowHelper.ExceptionResource.PinNotOpenedError, pin: pinNumber);
            }
            Interop.SetGpiodLineValue(pinHandle, (value == PinValue.High) ? 1 : 0);
           
```